example-app/src/main/java/backtraceio/backtraceio/MainActivity.java

```diff
@@ -33,9 +43,37 @@ protected void onCreate(Bundle savedInstanceState) {
         settings.setRetryOrder(RetryOrder.Queue);
 
         BacktraceDatabase database = new BacktraceDatabase(context, settings);
-        BacktraceClient backtraceClient = new BacktraceClient(context, credentials, database);
+        backtraceClient = new BacktraceClient(context, credentials, database);
 
         BacktraceExceptionHandler.enable(backtraceClient);
         backtraceClient.send("test");
+
+        // Enable handling of native crashes
+        database.setupNativeIntegration(backtraceClient, credentials);
+
+        // Enable ANR detection with a timeout of 3000 milliseconds
+        backtraceClient.enableAnr(3000);
+    }
+
+    public native void cppCrash();
+
+    public void handledException(View view) {
+        try {
```

Comment:
From demo perspective - throwing this kind of exceeptions won't give any cool insights about exception for user. I would recommend:
* For example - make a method that will do a file operation on files that don't exist 
* For example - trying to access nth index of array with n-1 elements. 

In this situation user will see:
* inner methods in MainActivity to see how easy is to find a root cause of problems,
* will see real exceptions that happen in every managed android app,
* will see more details in web debugger.

Reply:
I like this idea for more interesting exceptions well.

Reply:
I put some more interesting examples. 

The first is a handled exception for an equipment management interface, which is fairly typical for any game.

The second tries to read a save file (which is not there). But @konraddysput we know due to the existing inner exception issue we will not be able to catch the actual innermost exception for this case.